cmake/public/cuda.cmake

```diff
@@ -60,11 +60,15 @@ find_package(CUDAToolkit REQUIRED)
 
 cmake_policy(POP)
 
+# resolve symlinks: GH-108931
+file(REAL_PATH CUDA_INCLUDE_DIRS _inc_cuda)
+file(REAL_PATH CUDAToolkit_INCLUDE_DIR _inc_ctk)
```

Comment:
Maybe more explicit references to the modern `FindCUDAToolkit` and the deprecated `FindCUDA` would be appropriate? E.g. `_inc_find_cuda`, `_inc_finda_cudatoolkit`. Idk if the snake case names are consistent with the style otherwise used in pytorch